demos/http/http_demo_plaintext/http_demo_plaintext.c

```diff
@@ -0,0 +1,552 @@
+
+            if( *pTcpSocket == -1 )
+            {
+                continue;
+            }
+
+            pServerInfo = pIndex->ai_addr;
+
+            if( pServerInfo->sa_family == AF_INET )
+            {
+                /* IPv4 */
+                ( ( struct sockaddr_in * ) pServerInfo )->sin_port = netPort;
+                serverInfoLength = sizeof( struct sockaddr_in );
+                inet_ntop( pServerInfo->sa_family,
+                           &( ( struct sockaddr_in * ) pServerInfo )->sin_addr,
+                           resolvedIpAddr,
+                           sizeof( resolvedIpAddr ) );
+            }
+            else
+            {
+                /* IPv6 */
+                ( ( struct sockaddr_in6 * ) pServerInfo )->sin6_port = netPort;
+                serverInfoLength = sizeof( struct sockaddr_in6 );
+                inet_ntop( pServerInfo->sa_family,
+                           &( ( struct sockaddr_in * ) pServerInfo )->sin_addr,
+                           resolvedIpAddr,
+                           sizeof( resolvedIpAddr ) );
+            }
+
+            LogInfo( ( "Attempting to connect to resolved IP address: %s.",
+                       resolvedIpAddr ) );
+
+            returnStatus = connect( *pTcpSocket, pServerInfo, serverInfoLength );
+
+            if( returnStatus == -1 )
+            {
+                close( *pTcpSocket );
+            }
+            else
+            {
+                LogInfo( ( "Connected to IP address: %s.",
+                           resolvedIpAddr ) );
+                break;
+            }
+        }
+
+        if( pIndex == NULL )
+        {
+            /* Fail if no connection could be established. */
+            returnStatus = EXIT_FAILURE;
+            LogError( ( "Could not connect to any resolved IP address from %.*s.\n",
+                        ( int ) strlen( pServer ),
+                        pServer ) );
+        }
+        else
+        {
+            returnStatus = EXIT_SUCCESS;
+            LogInfo( ( "TCP connection established with %.*s.\n",
+                       ( int ) strlen( pServer ),
+                       pServer ) );
+        }
+    }
+    else
+    {
+        LogError( ( "Could not resolve host %.*s.\n",
+                    ( int ) strlen( pServer ),
+                    pServer ) );
+        returnStatus = EXIT_FAILURE;
+    }
+
+    /* Set the socket option for send and receive timeouts. */
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        transportTimeout.tv_sec = 0;
+        transportTimeout.tv_usec = ( TRANSPORT_SEND_RECV_TIMEOUT_MS * 1000 );
+
+        /* Set the receive timeout. */
+        if( setsockopt( *pTcpSocket,
+                        SOL_SOCKET,
+                        SO_RCVTIMEO,
+                        ( char * ) &transportTimeout,
+                        sizeof( transportTimeout ) ) < 0 )
+        {
+            LogError( ( "Setting socket receive timeout failed." ) );
```

Comment:
You can log the error code as well

Reply:
Hmm that's a little strange because the only other error code here is -1, and this is consistent with MQTT logs as well.

Reply:
We probably should log an error as there are supposedly more errors: https://pubs.opengroup.org/onlinepubs/009695399/functions/setsockopt.html